osquery/dispatcher/scheduler.cpp

```diff
@@ -100,7 +106,11 @@ SQLInternal monitor(const std::string& name, const ScheduledQuery& query) {
 
 Status launchQuery(const std::string& name, const ScheduledQuery& query) {
   // Execute the scheduled query and create a named query object.
-  VLOG(1) << "Executing scheduled query " << name << ": " << query.query;
+  if (FLAGS_verbose) {
+    VLOG(1) << "Executing scheduled query " << name << ": " << query.query;
+  } else if (FLAGS_schedule_lognames) {
```

Comment:
This may be surprising if the end user intended to have both. I'm not sure what people would expect if they provided both options

Reply:
True but this feels really nuanced. I cannot imagine why someone would want both. :/

Reply:
I can imagine consolodating logs, and diverting the output from `schedule_lognames` and info into some archival logstore. And occasionally using `verbose` to debug a problem. Ideally, you wouldn't want to lose the other stream of logs just because you're in verbose mode, verbose should be additive. (even if duplicative)

But, conversely, maybe most people are just running this at small scale, and accidentally getting 2x the output would confuse them.

I guess saying all that, I think it's more correct to split them. But I don't feel so strongly to raise it as an object, mostly a footnote. It's easy to change if people complain about it

Reply:
For my use case if I turned on verbose I would hope to also see the query itself. I agree with @directionless in the fact that verbose logs should be additive. My preference would be to have the duplicate the message over not having the query logged.  Although it's not necessarily adding a new line we could do:

```
VLOG(1) << "Executing scheduled query " << name << ": " << query.query;
if (FLAGS_schedule_lognames && !FLAGS_verbose) {
    LOG(INFO) << "Executing scheduled query " << name;
}
```
This way we maintain existing behavior but now also get the INFO line.

Reply:
Unless there's a specific reason to output to VLOG (which though in the end gets printed as if it was at INFO level), you could do something like this?

```
std::string optional_query;
if(FLAGS_schedule_lognames || FLAGS_verbose) {
  optional_query = ": " + query.query;
}

LOG(INFO) << "Executing scheduled query " << name << optional_query;
```

Reply:
You both have thought a lot more about this compared to myself. I like the analysis. I'll let you decide, I'm happy to accept and merge whatever you think is best :D

Reply:
@Breakwell It's your PR, you get to decide :) (I'd approve either way)

I think if you wanted verbose to replace, I'd leave as is, 

If you wanted additive, I'd go with separate if blocks
```
  if (FLAGS_verbose) {
    VLOG(1) << "Executing scheduled query " << name << ": " << query.query;
  }

if (FLAGS_schedule_lognames) {
    LOG(INFO) << "Executing scheduled query " << name;
  }
```

Reply:
I am fine with it as it is.

Reply:
Okay! I want to give @theopolis a chance to re-review, since he flagged it as needing changes. But I expect we can merge this very soon